lib/records/unit-tests/main.cpp

```diff
@@ -0,0 +1,25 @@
+/** @file
+    
+  This file used for catch based tests. It is the main() stub.
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#define CATCH_CONFIG_MAIN 
+#include "catch.hpp"
```

Comment:
It seems to me that compiling this in the `check` target is taking a significant amount of time. We may want to put this in a common place and reuse the object from there.

Reply:
Yes. There was an effort originally to put all tests in one compilation unit for this reason but it had other problems. For now it's supposed to be done per library basically (e.g. all the librecords based unit tests should be in one compilation unit).